File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.py
from collections import deque
def solution(food):
    answer = '0'
    N = len(food)
    # 1번 인덱스부터 홀수개면 -1개
    for i in range(N-1, 0, -1):
        cnt = food[i]
        if (cnt % 2 == 1):
            cnt -= 1
            
        cnt //= 2
        answer = str(i) * cnt + answer + str(i) * cnt
        
    # 1223330333221
    return answer

File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.java
class Solution {
    public String solution(int[] food) {
        String answer = "0";
        int N = food.length;
        
        for (int i = N-1; i > 0; i--) {
            int cnt = food[i];
            if (cnt % 2 == 1) {
                cnt--;
            }
            cnt /= 2;
            
            String tmp = "";
            for (int j = 0; j < cnt; j++) {
                tmp += Integer.toString(i);
            }
            answer = tmp + answer + tmp;
        }
        return answer;
    }
}